actions/base/action.py

```diff
@@ -79,3 +83,126 @@ def _get_cache(self):
+                    "'/opt/stackstorm/packs/msexchange/attachments'.")
+                self.logger.error(err_msg)
+                attach_dir = "/opt/stackstorm/packs/msexchange/attachments"
+        else:
+            attach_dir = os.path.abspath(attach_dir)
+
+        # Create the folder/directory, if it doesn't exist,
+        # and make it writeable.
+        if not os.path.exists(attach_dir):
+            os.makedirs(attach_dir, exist_ok=True)
+            os.chmod(attach_dir, 0o755)
+            self.logger.info("Created directory '{dir}' and made writeable."
+                             .format(dir=attach_dir))
+
+        if not os.access(attach_dir, os.W_OK):
+            raise OSError("Unable to write to attachment directory '{dir}'."
+                          .format(dir=attach_dir))
+
+        self.attachment_directory = attach_dir
+        self.attachment_directory_maximum_size = int(self.config.get(
+            "attachment_directory_maximum_size", 50))
+        self.attachment_days_to_keep = int(self.config.get(
+            "attachment_days_to_keep", 7))
+
+    def _get_date_from_string(self, date_str=None):
+        """
+        Use dateutil library (https://dateutil.readthedocs.io/) to parse
+        unstructured date string to standard format.
+        :param date_str str: Date as string in unknown/unstructured format
+        :returns EWSDateTime object or None
+        """
+        # If date_str is not provided, we assume that this is for the *end*
+        # of the filter range, which we set to "now", using timezone from
+        # pack configuration.
+        if not date_str:
+            return EWSDateTime.now(tz=self.timezone)
+
+        try:
+            from dateutil import parser
+            import pytz
+            parsed_date = parser.parse(date_str)
+            utc_date = pytz.utc.localize(parsed_date)
+            local_date = utc_date
+            try:
+                local_date = utc_date.astimezone(self.timezone)
+            except Exception:
+                self.logger.error("Unable to convert search date to pack "
+                                  "timezone. Using UTC...")
+            start_date = EWSDateTime.from_datetime(local_date)
+            self.logger.debug("Search start date: {dt}".format(dt=start_date))
+        except ImportError:
+            self.logger.error("Unable to find/load 'dateutil' library.")
+            start_date = None
+        except ValueError:
+            self.logger.error("Invalid format for date input: {dt}"
+                              .format(dt=date_str))
+            start_date = None
+
+        return start_date
+
+    def _search_items(self, folder, subject=None, search_start_date=None):
+        """
+        Common method for searching for MS Exchange items (email messages,
+        calendar items, etc.). Used by _search_items_ and _save_attachments_.
+        """
+        folder = self.account.root.get_folder_by_name(folder)
```

Comment:
Maybe to prep for the upgrade of exchangelib  in #16, remove the `get_folder_by_name` outlined in #20 as it is now deprecated.